pvlib/irradiance.py

```diff
@@ -1198,6 +1198,48 @@ def perez(surface_tilt, surface_azimuth, dhi, dni, dni_extra,
         return sky_diffuse
 
 
+def clearsky_index(ghi, clearsky_ghi, max_clearsky_index=2.0):
+    """
+    Calculate the clearsky index.
+
+    The clearsky index is the ratio of global to clearsky global irradiance.
+    Negative and non-finite clearsky index values will be truncated to zero.
+
+    Parameters
+    ----------
+    ghi : numeric
+        Global horizontal irradiance in W/m^2.
+
+    clearsky_ghi : numeric
+        Modeled clearsky GHI
+
+    max_clearsky_index : numeric, default 2.0
+        Maximum value of the clearsky index. The default, 2.0, allows
+        for over-irradiance events typically seen in sub-hourly data.
+
+    Returns
+    -------
+    clearsky_index : numeric
+        Clearsky index
+    """
+    clearsky_index = ghi / clearsky_ghi
+    # set +inf, -inf, and nans to zero, while preserving input type
+    # but preserve nans in the input arrays as well
+    input_is_nan = ~np.isfinite(ghi) | ~np.isfinite(clearsky_ghi)
```

Comment:
why use `isfinite` instead of `isnan` here? To set input `inf` to `nan`? Seems like that is unnecessary and may be harmful in some weird situations.

Reply:
Using `isnan` will map input `inf` to a clearsky index of `0`.  It's not clear to me that `0` would be preferable to `nan` for `inf` inputs, but if you think that returning zero is better then that's fine by me.

Reply:
Ah, I see how that works now. It's fine to leave it as is. It's documented correctly. I suggest that if someone doesn't like it they can make an issue and fix it.